control/src/http/controller_impl.cc

```diff
@@ -39,14 +39,21 @@ std::shared_ptr<ServiceContext> ControllerImpl::GetServiceContext(
     return std::make_shared<ServiceContext>(client_context_,
                                             *config.legacy_config);
   }
-  auto config_map = client_context_->config().service_configs();
-  auto it = config_map.find(config.destination_service);
-  if (it == config_map.end()) {
-    it = config_map.find(
-        client_context_->config().default_destination_service());
+
+  const std::string& origin_name = config.destination_service;
```

Comment:
Where is cache cleared? Do we just create a new controller any time the client config changes?

Reply:
Yes.  if client config changed,  xDS will send new envoy filter config, and new controller will be created.